packages/core/src/services/documents/create.test.ts

```diff
@@ -58,4 +58,26 @@ describe('createNewDocument', () => {
       'Cannot create a document version in a merged commit',
     )
   })
+
+  it('modifies other documents if it is referenced by another document', async (ctx) => {
+    const { project } = await ctx.factories.createProject({
+      documents: {
+        main: '<ref prompt="referenced/doc" />',
+      },
+    })
```

Comment:
Added a new attribute to the project factory to start with an initial document structure :)